### PR TITLE
Basic gcs usage in testgrid updater

### PR DESCRIPTION
**Pull request by fejta**
This makes the testgrid dir list a bucket (mostly done to add these deps to the repo)

Also:
* Update `hack/update-deps.sh` to accept command line args and delete the testdata dir before doing bazel run
* Run `hack/update-deps.sh --add cloud.google.com/go/storage@v0.17.0`
* Run `hack/update-deps.sh --update golang.org/x/net golang.org/x/oauth2`

Please review commit individually as the generated changes from `hack/updated-deps.sh` is enormous

/assign @ixdy @michelle192837 @rmmh

### CODE REVIEW

File: testgrid/updater/main.go [hunk 1]
```diff
@@ -17,12 +17,27 @@ limitations under the License.
...
+	"cloud.google.com/go/storage"
 )
 
 func main() {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("Failed to create storage client: %v", err)
+	}
+	bkt := client.Bucket("kubernetes-jenkins")
```
Comment (ixdy) [inline testgrid/updater/main.go]: should the bucket be configurable, rather than hard-coded like this?

File: hack/update-deps.sh [hunk 2]
```diff
@@ -28,7 +28,8 @@ TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
 cd "${TESTINFRA_ROOT}"
 
 trap 'echo "FAILED" >&2' ERR
-bazel run //:dep -- ensure -v
+rm -rf vendor/github.com/golang/dep/internal/fs/testdata
+bazel run //:dep -- ensure -v "$@"
 # dep itself has a problematic testdata directory with infinite symlinks which
```
Comment (ixdy) [inline hack/update-deps.sh]: maybe move the comment up, too?

File: hack/update-deps.sh [hunk 3]
```diff
@@ -28,7 +28,8 @@ TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
 cd "${TESTINFRA_ROOT}"
 
 trap 'echo "FAILED" >&2' ERR
-bazel run //:dep -- ensure -v
+rm -rf vendor/github.com/golang/dep/internal/fs/testdata
+bazel run //:dep -- ensure -v "$@"
```
Comment (ixdy) [inline hack/update-deps.sh]: I was going to ask if the comment for the script needed updating for this functionality, but then I noticed it's describing `prune-libraries.sh`, not `update-deps.sh`.

in any case, maybe update that comment? probably worth updating `docs/dep.md` too.

Comment (fejta) [reply hack/update-deps.sh]: Done

File: hack/update-deps.sh (hunk 2 above)
Comment (fejta) [reply hack/update-deps.sh]: Done

File: testgrid/updater/main.go (hunk 1 above)
Comment (fejta) [reply testgrid/updater/main.go]: Uh, no.

**Comment by michelle192837** [comment]
Looks fine to me on TestGrid side

File: docs/dep.md [hunk 4]
```diff
@@ -1,40 +1,32 @@
...
+* [update] - Run `hack/update-deps.sh --update golang.org/x/net [...]` to update an existing dependency
+* [remove] - Edit `Gopkg.toml` and run `hack/update-deps.sh` to remove or unpin dependencies
 
 ## Tips
 
 If `dep ensure` doesn't come back and freezes, please make sure `hg` command is
 installed on your environment. `dep ensure` requires `hg` command for getting
 bitbucket.org/ww/goautoneg , but `dep ensure` doesn't output such error message
-and just freezes. [reference](https://github.com/kubernetes/test-infra/issues/5987)
+and just [freezes]
```
Comment (ixdy) [inline docs/dep.md]: nit: please add back the trailing period. :)

File: testgrid/updater/main.go (hunk 1 above)
Comment (ixdy) [reply testgrid/updater/main.go]: ok, I clearly don't understand what this is used for. :)

**Comment by ixdy** [comment]
besides one lingering minor nit (that `.`) this LGTM. feel free to self-lgtm after fixing.

File: testgrid/updater/main.go (hunk 1 above)
Comment (rmmh) [reply testgrid/updater/main.go]: This code that is being expanded in small commits. The fact that it doesn't *do* anything just yet means it's all implicitly `//TODO`. :-)

**Comment by rmmh** [comment]
Can't self-lgtm, so I'll do the next best thing:

/lgtm
/hold

File: docs/dep.md (hunk 4 above)
Comment (fejta) [reply docs/dep.md]: :roll_eyes:

Comment (fejta) [reply docs/dep.md]: Done

File: testgrid/updater/main.go (hunk 1 above)
Comment (fejta) [reply testgrid/updater/main.go]: I have more commits to add, but I'd rather get in the vendored code first so the rest are vastly easier to review. And this at least uses it so dep won't remove it

**Comment by fejta** [comment]
Jeff, any more thoughts?

**Comment by ixdy** [comment]
oh right, somehow I forgot about `/hold`.

**Pull request merged**